src/clusterfuzz/stacktraces/__init__.py

```diff
@@ -173,7 +179,8 @@ def update_state_on_match(self,
       state.frame_count = 0
 
     # Direct updates.
-    if new_type is not None:
+    if new_type is not None and self.get_rank(new_type) >= self.get_rank(
```

Comment:
Can you comment that we do `>=` because we want later types to be prioritized?

Reply:
Thanks, I added a comment explaining the prioritization.